docs/user_guide/selection/SmartCorrelatedSelection.rst

```diff
@@ -344,6 +351,143 @@ the dataframe:
 
     [True, True, True, True, False, True, False, True, False, False, True, True]
 
+
+Correlation with Target
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Finally, let's select the feature that shows the strongest correlation with the target
+variable from each group.
+
+Let's create another toy dataframe with 4 features and a target variable:
+
+.. code:: python
+
+    import pandas as pd
+    from sklearn.datasets import make_classification
+    from feature_engine.selection import SmartCorrelatedSelection
+
+    # make dataframe with some correlated variables
+    def make_data():
+        X, y = make_classification(n_samples=1000,
+                                   n_features=12,
+                                   n_redundant=4,
+                                   n_clusters_per_class=1,
+                                   weights=[0.50],
+                                   class_sep=2,
+                                   random_state=1)
+
+        # transform arrays into pandas df and series
+        colnames = ['var_'+str(i) for i in range(12)]
+        X = pd.DataFrame(X, columns=colnames)
+        return X, y
+
+    X, y = make_data()
+
+Now, we set up :class:`SmartCorrelatedSelection` to find features groups with absolute
+correlation coefficient >0.8 and retain the feature with the strongest correlation with
+the target variable:
+
+.. code:: python
+
+    # set up the selector
+    tr = SmartCorrelatedSelection(
+        variables=None,
+        method="pearson",
+        threshold=0.8,
+        missing_values="raise",
+        selection_method="corr_with_target",
+        estimator=None,
+    )
+
+With `fit()`, the transformer finds the correlated variables and selects the ones to
+keep. With `transform()`, it drops the remaining features in the correlated group from
+the dataset:
+
+.. code:: python
+
+    Xt = tr.fit_transform(X, y)
+
+Let's explore the correlated feature groups:
+
+.. code:: python
+
+    tr.correlated_feature_sets_
+
+Similarly to the previous examples, we see that the groups of correlated features are
+slightly different, because in this cases, the features were assessed in alphabetical
+order, whereas when we used the variance the features we sorted based on their standard
+deviation for the assessment.
+
+.. code:: python
+
+    [{'var_4', 'var_6', 'var_7', 'var_9'}, {'var_0', 'var_8'}]
+
+We can find the feature that will be retained as the key in the following attribute:
+
+.. code:: python
+
+    tr.correlated_feature_dict_
+
+The variables `var_7` and `var_0` will be retained, and the remaining ones will be dropped.
+
+.. code:: python
+
+    {'var_7': {'var_4', 'var_6', 'var_9'}, 'var_0': {'var_8'}}
+
```

Comment:
After this code, I think it would ne nice to show that the correlation of var_0 with the target is higher than the correlation of var_8 with the target. Would you mind adding that @ClaudioSalvatoreArcidiacono ?